### PR TITLE
Chore: Fix lint script and issues

### PR DESCRIPTION
**Pull request by nrabinowitz**
As far as I can tell, `yarn lint` has not been checking files in `src` since [this change](https://github.com/keplergl/kepler.gl/commit/10868ecfb371e7547bf63daa205bdd62ebdd2c29#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R63). This fixes the `yarn lint` script and addresses the lint failures it reveals.

Some of these issues can be resolved by upgrading `eslint`, `prettier`, and/or `eslint-plugin-prettier`, but for now this just gets lint green.

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -43,6 +43,14 @@
...
     "comma-dangle": 1
   },
+  "overrides": [
+    {
+      "files": ["*.ts", "*.tsx"],
+      "rules": {
+        "no-undef": "off",
+      },
+    },
+  ],
```
Comment (nrabinowitz) [inline .eslintrc:53]: See https://github.com/typescript-eslint/typescript-eslint/blob/main/docs/linting/TROUBLESHOOTING.md#i-get-errors-from-the-no-undef-rule-about-global-variables-not-being-defined-even-though-there-are-no-typescript-errors

File: package.json [hunk 2]
```diff
@@ -61,7 +61,7 @@
     "typedoc": "typedoc --theme markdown --out typedoc --inputFiles ./src/reducers --inputFiles ./src/actions --excludeExternals --excludeNotExported --excludePrivate",
     "example-version": "babel-node ./scripts/edit-version.js",
     "prettier-all": "prettier --write '{src,examples,test}/**/*.{js,tsx,ts}'",
-    "lint": "yarn typescript && eslint src/*/src/*.{js,tsx,ts}  test webpack examples/*.js examples/**/src/*.{js,tsx,ts} website/*.js website/src --fix",
+    "lint": "yarn typescript && eslint src/*.js src/**/*.{js,tsx,ts} test webpack examples/*.js examples/**/src/*.{js,tsx,ts} website/*.js website/src --fix",
```
Comment (nrabinowitz) [inline package.json:64]: Note that for some reason just using `src` here caused `yarn lint` to run for several hours before I killed it - not sure why, but this seemed preferable.

**Comment by nrabinowitz** [comment]
The build failed lint with 220 additional errors, which I don't understand. Need to investigate why my local env is so different from CI, and then decide whether it's worth fixing all those errors now.

**Review by jwasilgeo — APPROVED** [review]
I was able to `yarn`, `yarn lint`, and `yarn start` without any issues that I could identify. LGTM

**Review by heshan0131 — APPROVED** [review]

**Pull request merged**